app/src/main/java/com/mstudio/superheroarch/MainActivity.kt

```diff
@@ -18,8 +26,29 @@ class MainActivity : AppCompatActivity() {
             v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
             insets
         }
+
+        val retrofit = Retrofit.Builder()
+            .baseUrl("https://rickandmortyapi.com/api/")
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
         findViewById<Button>(R.id.mainButton).setOnClickListener {
-            findViewById<TextView>(R.id.mainTextView).text = getString(R.string.changed_string)
+            CoroutineScope(Dispatchers.IO).launch {
+                val response = retrofit.create(RickAndMortyApiService::class.java).getCharacters()
+
+                if (response.isSuccessful) {
+                    val characterList = response.body()
+                    runOnUiThread {
```

Comment:
why is this needed here and not in the error case where you show a snackbar? Are snackbars thread-safe?

Reply:
seems like 😄